instrumentation/jetty-httpclient/jetty-httpclient-9.2/library/src/main/java/io/opentelemetry/javaagent/instrumentation/jetty/httpclient/v9_2/internal/JettyHttpClient9TracingInterceptor.java

```diff
@@ -0,0 +1,164 @@
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.client.api.Result;
+import org.eclipse.jetty.http.HttpField;
+import org.eclipse.jetty.http.HttpHeader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * JettyHttpClient9TracingInterceptor does three jobs stimulated from the Jetty Request object from
+ * attachToRequest() 1. Start the CLIENT span and create the tracer 2. Set the listener callbacks
+ * for each important lifecycle actions that would cause the start and close of the span 3. Set
+ * callback wrappers on two important request-based callbacks
+ */
+public class JettyHttpClient9TracingInterceptor
+    implements Request.BeginListener,
+        Request.FailureListener,
+        Response.SuccessListener,
+        Response.FailureListener,
+        Response.CompleteListener {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(JettyHttpClient9TracingInterceptor.class);
+
+  @Nullable private Context context;
+
+  @Nullable
+  public Context getContext() {
+    return this.context;
+  }
+
+  private final Context parentContext;
+
+  private final Instrumenter<Request, Response> instrumenter;
+
+  public JettyHttpClient9TracingInterceptor(
+      Context parentCtx, Instrumenter<Request, Response> instrumenter) {
+    this.parentContext = parentCtx;
+    this.instrumenter = instrumenter;
+  }
+
+  public void attachToRequest(Request jettyRequest) {
+    List<JettyHttpClient9TracingInterceptor> current =
+        jettyRequest.getRequestListeners(JettyHttpClient9TracingInterceptor.class);
+
+    if (!current.isEmpty()) {
+      LOG.warn("A tracing interceptor is already in place for this request! ");
+      return;
+    }
+    startSpan(jettyRequest);
+
+    // wrap all important request-based listeners that may already be attached, null should ensure
+    // are returned here
+    List<Request.RequestListener> existingListeners = jettyRequest.getRequestListeners(null);
+    wrapRequestListeners(existingListeners);
+
+    jettyRequest
+        .onRequestBegin(this)
+        .onRequestFailure(this)
+        .onResponseFailure(this)
+        .onResponseSuccess(this);
+  }
+
+  private void wrapRequestListeners(List<Request.RequestListener> requestListeners) {
+
+    List<Request.BeginListener> beginListeners =
+        requestListeners.stream()
+            .filter(rl -> rl instanceof Request.BeginListener)
+            .map(rl -> (Request.BeginListener) rl)
+            .collect(Collectors.toList());
+    List<Request.FailureListener> failureListeners =
+        requestListeners.stream()
+            .filter(rl -> rl instanceof Request.FailureListener)
+            .map(rl -> (Request.FailureListener) rl)
+            .collect(Collectors.toList());
+
+    Context context = this.parentContext;
+    for (Request.BeginListener blOriginal : beginListeners) {
+      requestListeners.set(
+          requestListeners.indexOf(blOriginal),
```

Comment:
Looks like here too @mateuszrzeszutek's suggestion of using `ListIterator` was missed

Reply:
I think I have this converted now